contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,382 @@
+        if ($.expireTimes[ipId] < block.timestamp) {
+            revert Errors.LicenseRegistry__IpExpired(ipId);
+        }
+        $.attachedLicenseConfigs[ipId].add(licenseConfigId);
+    }
+
+    // solhint-disable-next-line code-complexity
+    function registerDerivativeIp(
+        address derivativeIpId,
+        address[] calldata originalIpIds,
+        address licenseTemplate,
+        uint256[] calldata licenseConfigIds
+    ) external onlyLicensingModule {
+        if (originalIpIds.length == 0) {
+            revert Errors.LicenseRegistry__NoOriginalIp();
+        }
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if ($.attachedLicenseConfigs[derivativeIpId].length() > 0) {
+            revert Errors.LicenseRegistry__DerivativeIpAlreadyHasLicense(derivativeIpId);
+        }
+        if ($.originalIps[derivativeIpId].length() > 0) {
+            revert Errors.LicenseRegistry__DerivativeAlreadyRegistered(derivativeIpId);
+        }
+
+        for (uint256 i = 0; i < originalIpIds.length; i++) {
+            if ($.disputeModule.isIpTagged(originalIpIds[i])) {
+                revert Errors.LicenseRegistry__OriginalIpTagged(originalIpIds[i]);
+            }
+            if (derivativeIpId == originalIpIds[i]) {
+                revert Errors.LicenseRegistry__DerivativeIsOriginal(derivativeIpId);
+            }
+            if ($.expireTimes[originalIpIds[i]] < block.timestamp) {
+                revert Errors.LicenseRegistry__OriginalIpExpired(originalIpIds[i]);
+            }
+            // derivativeIp can only register with default license terms or the same license terms as the original IP
+            if ($.defaultLicenseTemplate != licenseTemplate || $.defaultLicenseConfigId != licenseConfigIds[i]) {
+                if ($.licenseTemplates[originalIpIds[i]] != licenseTemplate) {
+                    revert Errors.LicenseRegistry__OriginalIpUnmachedLicenseTemplate(originalIpIds[i], licenseTemplate);
+                }
+                if (!$.attachedLicenseConfigs[originalIpIds[i]].contains(licenseConfigIds[i])) {
+                    revert Errors.LicenseRegistry__OriginalIpHasNoLicenseConfig(originalIpIds[i], licenseConfigIds[i]);
+                }
+            }
+            $.originalIps[derivativeIpId].add(originalIpIds[i]);
+            $.derivativeIps[originalIpIds[i]].add(derivativeIpId);
+            $.attachedLicenseConfigs[derivativeIpId].add(licenseConfigIds[i]);
+        }
+
+        $.licenseTemplates[derivativeIpId] = licenseTemplate;
+        _setExpireTime(
+            derivativeIpId,
+            ILicenseTemplate(licenseTemplate).getEarlierExpireTime(block.timestamp, licenseConfigIds)
+        );
+    }
+
+    function verifyMintLicenseToken(
+        address originalIpId,
+        address licenseTemplate,
+        uint256 licenseConfigId,
+        bool isMintedByIpOwner
+    ) external view returns (Licensing.MintingLicenseSpec memory) {
```

Comment:
This method should be callable only by licensing module

Reply:
As this is a view function can be called without restrictions.